sdk/storage/azure-storage-blobs/src/private/avro_parser.cpp

```diff
@@ -0,0 +1,680 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "avro_parser.hpp"
+
+#include <algorithm>
+#include <cstring>
+
+#include <azure/core/azure_assert.hpp>
+#include <azure/core/internal/json/json.hpp>
+
+namespace Azure { namespace Storage { namespace Blobs { namespace _detail {
+
+  namespace {
+    int64_t parseInt(AvroStreamReader::ReaderPos& data)
+    {
+      uint64_t r = 0;
+      int nb = 0;
+      while (true)
+      {
+        uint8_t c = (*data.BufferPtr)[data.Offset++];
+        r = r | ((static_cast<uint64_t>(c) & 0x7f) << (nb * 7));
+        if (c & 0x80)
+        {
+          ++nb;
```

Comment:
I am confused as why we are moving 7 digits per iteration, e.g. for -128, it has only 1 byte, however, it is reading to second iteration, which reads another byte from the buffer.

could you add some test cases for negative numbers?

Reply:
Look at the binary encoding section at this doc https://avro.apache.org/docs/current/spec.html

Reply:
> could you add some test cases for negative numbers?

We don't test internal functions.

Integer parsing is a very basic functionality. If the whole avro parser works, I assume this function is good.

Reply:
My concern is that this is actually tested for positive numbers but not negative numbers. If you have confidence that's fine, but wouldn't hurt for some extra coverage.